### DISCUSSION TITLE
Implement Temporal

### DISCUSSION THREAD

**Discussion by dgrammatiko** [joomla/joomla-cms#32654] [2. Cold Features]
So, Javascript, the language, gets a new version every year since 2015. Most of the times the additions are some new features that have a minor impact on Joomla. A major exception here is the Intl and the temporal, which actually need an action plan.
In short, Intl allows javascript to format numbers, dates, etc based on the local language (either of the user's browser or predefined) and temporal is a new Date object. For further info, the PRs in the ecmascript repo are Intl: https://github.com/tc39/proposal-intl-locale and temporal: https://github.com/tc39/proposal-temporal

Temporal should be a first priority for the project as it enables support for all known calendars: https://github.com/tc39/proposal-temporal/blob/3fe5d062f58e2aa01e016b14c34bf0881ba2fb31/polyfill/lib/calendar.mjs#L1954-L1969

To be clear here, temporal will be soon part of ALL browsers but since there is an official Polyfill I propose a team (or someone with some advanced JS knowledge) to implement it in a progressive way (if the browser doesn't support it load the polyfill).
Supporting all the known calendars ootb is a far greater feature than some of the proposed GSOC projects that were announced a few days ago, so please prioritise tasks wisely.

Thanks

**Comment by brianteeman** [comment]
Oh cool. I came across temporal the other day working on that hebrew calendar you helped me with
